src/cuga/backend/server/manage_routes/config_routes.py

```
@@ -0,0 +1,483 @@
+        # Strip secrets from the PUBLISHED snapshot only — the draft (saved
+        # above) keeps the key so the local engine survives a restart.
+        # Importing instances on other machines must supply their own keys.
+        try:
+            from cuga.backend.knowledge.config import KnowledgeConfig as _KC_strip
+
+            published_config = dict(config or {})
+            kb_pub = published_config.get("knowledge")
+            if isinstance(kb_pub, dict):
+                kb_pub = dict(kb_pub)
+                for _secret in _KC_strip._SECRET_FIELDS:
+                    if _secret in kb_pub and kb_pub[_secret]:
```

Ruff (0.15.20)

[warning] Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)